solr/core/src/java/org/apache/solr/response/DocsStreamer.java

```diff
@@ -142,6 +142,11 @@ public boolean hasNext() {
     return docIterator.hasNext();
   }
 
+  // called at the end of the stream
```

Comment:
can't we leverage Closeable here and get some sugar&warns? 
Also, line 89 still calls setContext() .. is it right? or I'm missing something?